frontend-implicaction/src/app/discussion/components/top-group-listing/top-group-listing.component.ts

```diff
@@ -48,4 +50,37 @@ export class TopGroupListingComponent implements OnInit {
         width: 650
       });
   }
+
+  joinGroup(groupName: string): void {
+    this.userService.subscribeGroup(groupName)
+      .subscribe(
+        (group) => {
+          this.toasterService.success('Succès', `Vous avez adhéré au groupe ${group.name}`);
+          this.loadData();
+        },
```

Comment:
Si on considère la modif que j'ai proposée dans le back, plutôt que de renvoyé un groupe unique on renvoit la liste des groupes de l'utilisateur :

```suggestion
      .subscribe(
        groups => {
          this.toasterService.success('Succès', `Vous avez adhéré au groupe ${group.name}`);
          this.userGroups = groups;
        },
```